MAPL_Base/MAPL_GenericCplComp.F90

```diff
@@ -1103,4 +1122,437 @@ subroutine Finalize(CC, SRC, DST, CLOCK, RC)
+    integer                       :: STATUS
+
+! Locals
+
+    type (MAPL_GenericCplState), pointer  :: STATE
+    type (MAPL_GenericCplWrap )           :: WRAP
+    type(ESMF_VM)                         :: VM
+    type(ESMF_Grid)                       :: grid
+    type(ESMF_Field)                      :: field
+    character(len=ESMF_MAXSTR)            :: name
+    character(len=ESMF_MAXSTR)            :: filename
+    logical                               :: am_i_root
+    logical                               :: local_undefs
+    integer                               :: unit
+    integer                               :: n_vars
+    integer                               :: n_count
+    integer                               :: n_undefs
+    integer                               :: rank
+    integer                               :: i
+    integer                               :: dims
+    integer                               :: have_undefs
+    integer, pointer                      :: mask(:) => null()
+    real, allocatable                     :: buf1(:), buf2(:,:), buf3(:,:,:)
+    real, pointer                         :: ptr1(:), ptr2(:,:), ptr3(:,:,:)
+
+! Begin...
+
+! Get the target components name and set-up traceback handle.
+! -----------------------------------------------------------
+
+    IAm = "MAPL_GenericCplComWriteRestart"
+    call ESMF_CplCompGet( CC, NAME=COMP_NAME, RC=STATUS )
+    _VERIFY(STATUS)
+    Iam = trim(COMP_NAME) // Iam
+
+! Retrieve the pointer to the internal state. It comes in a wrapper.
+! ------------------------------------------------------------------
+
+    call ESMF_CplCompGetInternalState ( CC, WRAP, STATUS )
+    _VERIFY(STATUS)
+
+    STATE     =>  WRAP%INTERNAL_STATE
+
+    call ESMF_CplCompGet(CC, vm=vm, rc=status)
+    _VERIFY(STATUS)
+
+    filename = trim(state%name) // '_chk' ! following Andrea's suggestion
+    am_i_root = MAPL_AM_I_ROOT(vm)
+
+    unit=0 ! just to initialize
+    n_vars = size(state%src_spec)
+    if (am_i_root) then
+       UNIT = GETFILE(filename, rc=status)
+       _VERIFY(status)
+       write(unit) n_vars
+    end if
+
+       ! for each var
+    do i = 1, n_vars
+       ! varname we can get from query SHORT_NAME in state%src_spec(i)
+       call MAPL_VarSpecGet(state%src_spec(i), SHORT_NAME=name, rc=status)
+       _VERIFY(status)
+       call ESMF_StateGet(SRC, name, field=field, rc=status) 
+       _VERIFY(status)
+       call ESMF_FieldGet(field, grid=grid, rc=status)
+       _VERIFY(status)
+
+       rank = state%accum_rank(i)
+       call ESMF_AttributeGet(field, name='DIMS', value=DIMS, rc=status)
+       _VERIFY(STATUS)
+       mask => null()
+       if (DIMS == MAPL_DimsTileOnly .or. DIMS == MAPL_DimsTileTile) then
+          call MAPL_TileMaskGet(grid,  mask, rc=status)
+          _VERIFY(STATUS)
+       end if
+       ! ALT note: calling a procedure with optional argument, and passing NULL pointer to indicate "absent", needs ifort16 or newer
```

Comment:
For new code, I don't think this comment is necessary.  We're not backporting to SLES 11, and ifort 15 is not available on SLES 12.

Reply:
I agree. The comment was mostly a reminder for me. Plus, the original work was done for S2Sv3. I'll remove the new comments